aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -56,6 +56,26 @@ Tensor& bitwise_not_out(Tensor& result, const Tensor& self) {
   return result;
 }
 
+Tensor neg(const Tensor& self) {
+  Tensor result = at::empty({0}, self.options());
+  return at::neg_out(result, self);
+}
+
+Tensor& neg_(Tensor& self) {
+  return at::neg_out(self, self);
+}
+
+Tensor& neg_out(Tensor& result, const Tensor& self) {
+  checkBackend("neg", result, self.type().backend());
```

Comment:
I don't think you need checkBackend. TensorIterator should do those checks.

Reply:
OK, deleted